worker/runtime/spec/namespaces.go

```diff
@@ -13,14 +17,26 @@ var (
 
 	UnprivilegedContainerNamespaces = append(PrivilegedContainerNamespaces,
 		specs.LinuxNamespace{Type: specs.UserNamespace},
-		specs.LinuxNamespace{Type: specs.CgroupNamespace},
 	)
 )
 
+func init() {
+	cgroupNamespacesEnabled()
```

Comment:
Nit: would it be possible to rename this to something more verbose like `enableCgroupNamespacesIfSupported`? `cgroupNamespacesEnabled` doesn't suggest to me that there is mutation, and `init` already makes things a bit cryptic, so making it clear there's a mutation is probably a good thing.

Alternatively, it could be something like:
```go
func init() {
	if cgroupNamespacesSupported() {
		UnprivilegedContainerNamespaces = append(UnprivilegedContainerNamespaces,
			specs.LinuxNamespace{Type: specs.CgroupNamespace})
	}
}
```

which is even more clear IMO